source/backup-cursor.rst

```diff
@@ -0,0 +1,95 @@
+$backupCursor and $backupCursorExtended aggregation stages
+============================================================
+
+``$backupCursor`` and ``$backupCursorExtended`` aggregation stages allow developing your custom backup solution: 
+
+* ``$backupCursor`` outputs the list of files and their size to copy.  
+* ``$backupCursorExtended`` outputs the list of WiredTiger transaction log files that have been updated or newly added since the ``$backupCursor`` was first run. Saving these files enables restoring the database to any arbitrary time between the ``$backupCursor`` and ``$backupCursorExtend`` execution times.
+
+They are available in |PSMDB| starting with version 4.4.6-8. 
+
+Percona provides Percona Backup for MongoDB (PBM) – a light-weight open source solution for consistent backups and restores across sharded clusters. PBM relies on these aggregation stages for physical backups and restores. However, if you wish to develop your own backup application, this document describes the ``$backupCursor`` and ``$backupCursorExtended`` aggregation stages.
+
+Usage 
+=============
+
+Run both aggregation stages on one node from each replica set. You can use either the primary or the secondary node. Note that since the secondary node may lag in syncing the data from the primary one, you will have to wait for the exact same time before running the ``$backupCursorExtended``.
+
+Get a list of all files to copy with $backupCursor
+---------------------------------------------------------------
+
+.. code-block:: javascript
+
+    var bkCsr = db.getSiblingDB("admin").aggregate([{$backupCursor: {}}])
+    var bkCsrMetadata = bkCsr.next().metadata
+
+Sample output
+
+.. code-block:: 
+
+   bkCsr
+   [
+     {
+       metadata: {
+         backupId: UUID("35c34101-0107-44cf-bdec-fad285e07534"),
+         dbpath: '/var/lib/mongodb',
+         oplogStart: { ts: Timestamp({ t: 1666631297, i: 1 }), t: Long("-1") },
+         oplogEnd: { ts: Timestamp({ t: 1666631408, i: 1 }), t: Long("1") },
+         checkpointTimestamp: Timestamp({ t: 1666631348, i: 1 })
+       }
+     },
+     {
+       filename: '/var/lib/mongodb/journal/WiredTigerLog.0000000005',
+       fileSize: Long("104857600")
+     },
+     {
+       filename: '/var/lib/mongodb/WiredTigerHS.wt',
+       fileSize: Long("20480")
+     },
+     ....
+
+Store the ``metadata`` document somewhere, because you need to pass the ``backupId`` parameter from this document as the input parameter for the ``$backupCursorExtended`` stage. Also you need the ``oplogEnd`` timestamp.
+Make sure that the ``$backupCursor`` is complete on all shards in your cluster. 
+
+Run ``$backupCursorExtended`` to retrieve the WiredTiger transaction logs
+--------------------------------------------------------------------------
+
+.. code-block:: javascript
+
+   var bkExtCsr = db.aggregate([{$backupCursorExtend: {backupId: bkCsrMetadata.backupId, timestamp: new Timestamp(1666631418, 1)}}])
```

Comment:
Same here - remove var so that user gets output which is cool for educational purposes, IMHO)

bkExtCsr = db.aggregate([{$backupCursorExtend: {backupId: bkCsrMetadata.backupId, timestamp: new Timestamp(1666944242, 1)}}])
[
  {
    filename: '/home/vagrant/percona-server-mongodb-6.0.2-1-x86_64.glibc2.17/nodes/node1/db/journal/WiredTigerLog.0000000011'
  },
  {
    filename: '/home/vagrant/percona-server-mongodb-6.0.2-1-x86_64.glibc2.17/nodes/node1/db/key.db/WiredTigerLog.0000000009'
  },
  {
    filename: '/home/vagrant/percona-server-mongodb-6.0.2-1-x86_64.glibc2.17/nodes/node1/db/key.db/WiredTigerLog.0000000010'
  },
  {
    filename: '/home/vagrant/percona-server-mongodb-6.0.2-1-x86_64.glibc2.17/nodes/node1/db/key.db/WiredTigerLog.0000000007'
  },
  {
    filename: '/home/vagrant/percona-server-mongodb-6.0.2-1-x86_64.glibc2.17/nodes/node1/db/key.db/WiredTigerLog.0000000008'
  },
  {
    filename: '/home/vagrant/percona-server-mongodb-6.0.2-1-x86_64.glibc2.17/nodes/node1/db/key.db/WiredTigerLog.0000000011'
  }
]